python/paddle/fluid/tests/unittests/test_optimizer.py

```diff
@@ -614,7 +614,7 @@ def test_lookahead_optimizer(self):
 
 
 class TestRecomputeOptimizer(unittest.TestCase):
-    def net(self, return_input=False):
+    def net(self, return_input=False, with_dropout=False):
```

Comment:
How do you make sure a program has dropout? Do you scan all the ops?

Reply:
Yes. If there is no dropout in a program, the forward net will keep unchanged; if there are dropout ops in a program, it will be replaced with seed op and dropout op as shown in `backward.py`.